integration/spark/app/src/test/java/io/openlineage/spark/agent/column/ColumnLineageWithTransformationTypesTest.java

```diff
@@ -0,0 +1,322 @@
+            .sparkContext(spark.sparkContext())
+            .openLineage(new OpenLineage(Versions.OPEN_LINEAGE_PRODUCER_URI))
+            .queryExecution(queryExecution)
+            .meterRegistry(new SimpleMeterRegistry())
+            .openLineageConfig(new SparkOpenLineageConfig())
+            .build();
+
+    FileSystem.get(spark.sparkContext().hadoopConfiguration())
+        .delete(new Path("/tmp/column_non_v2/"), true);
+
+    spark.sql("DROP TABLE IF EXISTS t1");
+    spark.sql("DROP TABLE IF EXISTS t2");
+    spark.sql("DROP TABLE IF EXISTS t3");
+    spark.sql("DROP TABLE IF EXISTS t");
+  }
+
+  @Test
+  void simpleQueryOnlyIdentity() {
+    createTable("t1", "a;int");
+    OpenLineage.ColumnLineageDatasetFacet facet =
+        getFacetForQuery(getSchemaFacet("a;int"), "SELECT a FROM t1");
+
+    assertColumnDependsOnType(
+        facet, "a", FILE, T1_EXPECTED_NAME, "a", TransformationInfo.identity());
+  }
+
+  @Test
+  void simpleQueryOnlyTransform() {
+    createTable("t1", "a;int");
+    OpenLineage.ColumnLineageDatasetFacet facet =
+        getFacetForQuery(getSchemaFacet("a;int"), "SELECT concat(a, 'test') AS a FROM t1");
+
+    assertColumnDependsOnType(
+        facet, "a", FILE, T1_EXPECTED_NAME, "a", TransformationInfo.transformation());
+  }
+
+  @Test
+  void simpleQueryOnlyAggregation() {
+    createTable("t1", "a;int");
+    OpenLineage.ColumnLineageDatasetFacet facet =
+        getFacetForQuery(getSchemaFacet("a;int"), "SELECT count(a) AS a FROM t1");
+
+    assertColumnDependsOnType(
+        facet, "a", FILE, T1_EXPECTED_NAME, "a", TransformationInfo.aggregation());
+  }
+
+  @Test
+  void simpleQueryIndirect() {
+    createTable("t1", "a;int", "b;int");
+    OpenLineage.ColumnLineageDatasetFacet facet =
+        getFacetForQuery(getSchemaFacet("a;int"), "SELECT a FROM t1 WHERE b > 1");
+
+    assertColumnDependsOnType(
+        facet, "a", FILE, T1_EXPECTED_NAME, "a", TransformationInfo.identity());
+
+    assertColumnDependsOnType(
+        facet, "a", FILE, T1_EXPECTED_NAME, "b", TransformationInfo.indirect("FILTER"));
+  }
+
+  @Test
+  void simpleQueryMultipleIndirect() {
+    createTable("t1", "a;int", "b;int", "c;int");
+    OpenLineage.ColumnLineageDatasetFacet facet =
+        getFacetForQuery(
+            getSchemaFacet("a;int"), "SELECT a FROM t1 WHERE b > 1 GROUP BY a, c ORDER BY c");
+
+    assertColumnDependsOnType(
+        facet, "a", FILE, T1_EXPECTED_NAME, "a", TransformationInfo.identity());
+
+    assertColumnDependsOnType(
+        facet, "a", FILE, T1_EXPECTED_NAME, "a", TransformationInfo.indirect("GROUP_BY"));
+    assertColumnDependsOnType(
+        facet, "a", FILE, T1_EXPECTED_NAME, "b", TransformationInfo.indirect("FILTER"));
+    assertColumnDependsOnType(
+        facet, "a", FILE, T1_EXPECTED_NAME, "c", TransformationInfo.indirect("GROUP_BY"));
+    assertColumnDependsOnType(
+        facet, "a", FILE, T1_EXPECTED_NAME, "c", TransformationInfo.indirect("SORT"));
```

Comment:
I like the test where column depends on other column with two different types.

Reply:
and for indirect transformations you will probably get a lot of those